sources/engine/Xenko.Voxels/Voxels/Voxelization/VoxelVolumeProcessor.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xenko.Engine;
+using Xenko.Games;
+using Xenko.Extensions;
+using Xenko.Core.Mathematics;
+using Xenko.Graphics;
+using Xenko.Graphics.GeometricPrimitives;
+using Xenko.Rendering;
+using Xenko.Rendering.Voxels;
+using Xenko.Rendering.Shadows;
+using Xenko.Rendering.Materials;
+using Xenko.Rendering.Materials.ComputeColors;
+
+namespace Xenko.Engine.Processors
+{
+    public class VoxelVolumeProcessor : EntityProcessor<VoxelVolumeComponent>, IEntityComponentRenderProcessor
+    {
+        private Dictionary<VoxelVolumeComponent, DataVoxelVolume> renderVoxelVolumes = new Dictionary<VoxelVolumeComponent, DataVoxelVolume>();
+        public Dictionary<VoxelVolumeComponent, ProcessedVoxelVolume> processedVoxelVolumes = new Dictionary<VoxelVolumeComponent, ProcessedVoxelVolume>();
+        bool isDirty;
+        SceneSystem sceneSystem;
+        GraphicsDevice graphicsDevice;
+        CommandList commandList;
+
+        public VisibilityGroup VisibilityGroup { get; set; }
+        public RenderGroup RenderGroup { get; set; }
+        protected override void OnSystemAdd()
+        {
+            base.OnSystemAdd();
+
+            VisibilityGroup.Tags.Set(VoxelRenderer.CurrentRenderVoxelVolumes, renderVoxelVolumes);
+            VisibilityGroup.Tags.Set(VoxelRenderer.CurrentProcessedVoxelVolumes, processedVoxelVolumes);
+            VisibilityGroup.Tags.Set(VoxelRenderFeature.CurrentProcessedVoxelVolumes, processedVoxelVolumes);
+            sceneSystem = Services.GetService<SceneSystem>();
+            graphicsDevice = Services.GetService<IGraphicsDeviceService>().GraphicsDevice;
+            commandList = Services.GetService<CommandList>();
+        }
+        public override void Update(GameTime time)
+        {
+            RegenerateVoxelVolumes();
+        }
+        public override void Draw(RenderContext context)
+        {
+
```

Comment:
Yeah that makes sense. I based this off of the LightShaftProcessor where it's handled in Update, so I stuck with that since I knew it worked. Will move it to Draw.